CMakeLists.txt

```diff
@@ -43,418 +50,536 @@ include("cmake/ontologies.cmake")
+	src/queries/QueryContext.cpp
+	src/semweb/SPARQLQuery.cpp
+	src/semweb/TripleContainer.cpp
+	src/KnowRobError.cpp
+	src/integration/python/PythonError.cpp
+	src/storage/ReifiedTriple.cpp
+	src/storage/ReifiedQuery.cpp
+	src/storage/ReificationContainer.cpp
+	src/storage/UnReificationContainer.cpp
+	src/storage/StorageInterface.cpp
+	src/semweb/GraphUnion.cpp
+	src/semweb/GraphSequence.cpp
+	src/semweb/GraphTerm.cpp
+	src/storage/mongo/MongoTriple.cpp
+	src/storage/mongo/MongoTriplePattern.cpp
+	src/storage/mongo/MongoTerm.cpp
+	src/storage/mongo/MongoTaxonomy.cpp
+	src/storage/SPARQLBackend.cpp
+	src/queries/parsers/strings.cpp
+	src/queries/parsers/terms.cpp
+	src/queries/parsers/formula.cpp
+	src/reasoner/GoalDrivenReasoner.cpp
+	src/reasoner/DataDrivenReasoner.cpp
+	src/reasoner/ReasonerEvent.cpp
+	src/semweb/TransformedOntology.cpp
+	src/reasoner/Goal.cpp
+	src/formulas/SimpleConjunction.cpp
+	src/queries/ConjunctiveQuery.cpp
+	src/semweb/RDFIndicator.cpp
+	src/integration/python/converter/shared_ptr.cpp
+	src/semweb/GraphBuiltin.cpp
+	src/reasoner/RDFGoal.cpp
+	src/reasoner/RDFGoalReasoner.cpp
+	src/reasoner/Computable.cpp
+	src/storage/Observer.cpp
+	src/storage/ObserverManager.cpp
+	src/storage/ObserverJob.cpp
+	src/queries/parsers/graph.cpp)
+
+if(NOT KNOWROB_USE_STATIC_LIB)
+	set(KNOWROB_LIBRARIES
+		${SWIPL_LIBRARIES}
+		${MONGOC_LIBRARIES}
+		${RAPTOR2_LIBRARIES}
+		${REDLAND_LIBRARIES}
+		${Python3_LIBRARIES}
+		spdlog::spdlog
+		${GTEST_LIBRARIES})
+else()
+	pkg_check_modules(RAPTOR2 REQUIRED raptor2)
+	pkg_check_modules(XSLT REQUIRED libxslt)
+	pkg_check_modules(RASQAL REQUIRED rasqal)
+	pkg_check_modules(XML2 REQUIRED libxml-2.0)
+	pkg_check_modules(SPDLOG REQUIRED spdlog)
+	pkg_check_modules(CURL REQUIRED libcurl)
+	pkg_check_modules(SSL REQUIRED libssl)
+	pkg_check_modules(CRYPTO REQUIRED libcrypto)
+	pkg_check_modules(GMP REQUIRED gmp)
+	pkg_check_modules(YAJL REQUIRED yajl)
+	pkg_check_modules(UUID REQUIRED uuid)
+	pkg_check_modules(LDAP REQUIRED ldap)
+	pkg_check_modules(LBER REQUIRED lber)
+	pkg_check_modules(PCRE REQUIRED libpcre)
+	link_directories(${MONGOC_DIR}/../../lib/
+		${XSLT_LIBRARY_DIRS}
+		${RASQAL_LIBRARY_DIRS}
+		${XML2_LIBRARY_DIRS}
+		${SPDLOG_LIBRARY_DIRS}
+		${SSL_LIBRARY_DIRS}
+		${CURL_LIBRARY_DIRS}
+		${CRYPTO_LIBRARY_DIRS}
+		${GMP_LIBRARY_DIRS}
+		${YAJL_LIBRARY_DIRS}
+		${UUID_LIBRARY_DIRS}
+		${LDAP_LIBRARY_DIRS}
+		${LBER_LIBRARY_DIRS}
+		${PCRE_LIBRARY_DIRS})
+	set(KNOWROB_LIBRARIES
+		libraptor2.a
+		librdf.a
+		libxslt.a
+		librasqal.a
+		libxml2.a
+		liblzma.a
+		libswipl.a
+		libmongoc2.a
+		libsasl2.a
+		libbson2.a
+		libspdlog.a
+		libcurl.a
+		libssl.a
+		libcrypto.a
+		libgmp.a
+		libyajl_s.a
+		libuuid.a
+		libldap.a
+		liblber.a
+		libpcre.a
+		${Python3_LIBRARIES}
+		${GTEST_LIBRARIES}
+		resolv
+		zstd
+		ltdl
+		tinfo
+		fmt
+		dl
+		z
+		rt)
+	message(STATUS "KnowRob uses libraries: ${KNOWROB_LIBRARIES}")
+endif()
+
+set(OpenSSL_LIBRARIES "") # Unset to avoid leakage
```

Comment:
what leakage and where?

Reply:
no, it is not. there is no mention of "OpenSSL_LIBRARIES" in the current cmake

Reply:
Then I'm confused too :confused: I can remove it

Reply:
I thought it maybe was related to the problem we have with SSL plus raptor? i.e. the issue with gazebo. But at least in knowrob we do not use the library